### PR TITLE
feat(azure,dns): force usage of dns resource

### PR DESCRIPTION
**Pull request by JGodin-C2C**
Force the usage of azure_dns_zone
This should solve the dependency problem for dns resolution .

### CODE REVIEW

**Comment by ckaenzig** [comment]
Since the PR adds a mandatory variable to the module, it should be marked as a breaking change.

File: modules/aks/azure/variables.tf [hunk 1]
```diff
@@ -75,3 +75,12 @@ variable "node_pools" {
   description = "List of node pools with minimal configuration"
   type        = map(any)
 }
+
+variable "azurerm_dns_zone" {
+  description = "This is needed as the cluster needs to know its own DNS prefixes in order to deploy the needed application"
```
Comment (ckaenzig) [inline modules/aks/azure/variables.tf:80]: The description should describe the variable meaning, not so much why you added it.

**Comment by Vampouille** [comment]
> This should solve the dependency problem for dns resolution .

@JGodin-C2C Can you explain the problem with dependencies ?

**Comment by raphink** [comment]
@JGodin-C2C Could you update the docs as well please?

Note: I see there's no examples or tests for AKS. It would be good to add them at some point (cc @fsismondi @modridi )

**Comment by JGodin-C2C** [comment]
> > This should solve the dependency problem for dns resolution .
> 
> @JGodin-C2C Can you explain the problem with dependencies ?

When deploying the devops stack, if the resource does not exists, it will always fails as it tries to use something that should be already existing 

https://github.com/camptocamp/devops-stack/pull/822/files#diff-48b0e39c924c7640f3ad28f71082c6088db6d82c690a7a6052abd148469a631eL218

**Comment by JGodin-C2C** [comment]
> @JGodin-C2C Could you update the docs as well please?
> 
> Note: I see there's no examples or tests for AKS. It would be good to add them at some point (cc @fsismondi @modridi )

Test for AKS is ongoing. But this should go in anther PR.

**Comment by JGodin-C2C** [comment]
1.

**Pull request closed without merging**